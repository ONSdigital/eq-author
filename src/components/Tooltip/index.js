import React from "react";
import ReactTooltip from "react-tooltip";
import styled from "styled-components";
import PropTypes from "prop-types";
import { uniqueId } from "lodash";

const StyledTooltip = styled(ReactTooltip)`
    background-color: #707070 !important;
    &::before, &::after {
      border-bottom: none !important;
    }
`;

class Tooltip extends React.Component {
  getGeneratedId() {
    return this.id || (this.id = uniqueId("tooltip-"));
  }

  render() {
    const { children, content, ...otherProps } = this.props;
    const child = React.Children.only(children);
    const id = child.props.id || this.getGeneratedId();

    return (
      <div>
        {React.cloneElement(child, {
          "data-tip": true,
          "data-for": id
        })}
        <StyledTooltip id={id} place="bottom" effect="solid" {...otherProps}>
          {content}
        </StyledTooltip>
      </div>
    );
  }
}

Tooltip.propTypes = {
  content: PropTypes.node.isRequired,
  children: PropTypes.element.isRequired
};

export default Tooltip;
