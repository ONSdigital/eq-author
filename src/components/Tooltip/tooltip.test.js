import React from "react";
import { shallow } from "enzyme";
import Tooltip from "./index";

describe("Tooltip", () => {
  let component;

  beforeEach(() => {
    component = shallow(
      <Tooltip tooltipId="buttonTooltip" tooltipText="This is a button">
        <button>Click me</button>
      </Tooltip>
    );
  });

  it("should render", () => {
    expect(component).toMatchSnapshot();
  });
});
