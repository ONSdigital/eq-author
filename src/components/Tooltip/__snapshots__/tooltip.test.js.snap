// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip should render 1`] = `
<div>
  <button
    data-for="buttonTooltip"
    data-tip={true}
    id="buttonTooltip"
  >
    Click me
  </button>
  <Tooltip__StyledTooltip
    effect="solid"
    id="buttonTooltip"
    place="bottom"
  >
    This is a button
  </Tooltip__StyledTooltip>
</div>
`;

exports[`Tooltip should render arbitrary content 1`] = `
<div>
  <button
    data-for="buttonTooltip"
    data-tip={true}
    id="buttonTooltip"
  >
    Click me
  </button>
  <Tooltip__StyledTooltip
    effect="solid"
    id="buttonTooltip"
    place="bottom"
  >
    <ul>
      <li>
        foo
      </li>
      <li>
        bar
      </li>
    </ul>
  </Tooltip__StyledTooltip>
</div>
`;

exports[`Tooltip should use auto-generated id if one is not supplied 1`] = `
<div>
  <button
    data-for="tooltip-1"
    data-tip={true}
  >
    Click me
  </button>
  <Tooltip__StyledTooltip
    effect="solid"
    id="tooltip-1"
    place="bottom"
  >
    This is a tooltip
  </Tooltip__StyledTooltip>
</div>
`;
